internal/txapp/txapp.go

```diff
@@ -538,9 +552,8 @@ func (r *TxApp) ProposerTxs(ctx context.Context, txNonce uint64, maxTxSize int64
 		}
 
 		// MaxTxBytes restrictions per block are enforced here
-		// From observations, the rlp encoded events size is almost always smaller than the event
-		// so use eventsize for enfrocing maxTxSize and fill the rest of the blocksize with the mempool txns.
-		evtSz := eventSize(event)
+		// As the rlp encoded size is almost always smaller, enforcing it on the unencoded data for ease of use
+		evtSz := int64(len(event.Type)) + int64(len(event.Body))
 		if evtSz > maxTxSize {
 			break
 		}
```

Comment:
Do we want to `maxTxSize -= evtSz` whenever we `finalEvents = append(finalEvents, event)`?